akka-actor/src/main/scala/akka/util/ByteString.scala

```diff
@@ -124,29 +125,47 @@ object ByteString {
 
     override def length: Int = bytes.length
 
+    @deprecated("Avoid `iterator` in performance sensitive code, call ops directly on ByteString instead", "2.4.9")
```

Comment:
Open question, I guess. It helped me figure out where people call it.
I don't mind keeping the warning as `//` comment